shared/js/pageStore.js

```diff
@@ -100,6 +115,72 @@ loop.store.PageStore = function(mozL10n) {
           label: mozL10n.get("snackbar_page_deleted")
         }));
       }
+    },
+
+    /**
+     * Check whether this store already contains this URL.  Note that
+     * we currently use an exact string compare for this; we might want
+     * to consider switching to something less strict if it makes sense
+     * from a product standpoint.
+     *
+     * @param {String} url - the thing we're looking for
+     * @return {Boolean}
+     */
+    _hasPageForUrl(url) {
+      let pages = this._storeState.pages;
+      pages = pages.filter(page =>
+        page.url === url
+      );
+
+      if (pages.length === 1) {
+        return true;
+      }
+
+      // the store is corrupt
+      if (pages.length > 1) {
+        throw new Error("Page store contains mutiple pages with the same URL");
+      }
+
+      return false;
+    },
+
+    /**
+     * Handle updateRoomInfo action.  Used to add the first URL to the
+     * PageStore.
+     */
+    updateRoomInfo(actionData) {
+      if (!("roomContextUrls" in actionData)) {
+        console.warn("this room doesn't have a context url!");
+        return;
+      }
+
+      // XXX do we really care about multiple context urls?  I suspect
+      // we don't.
+      let firstContextUrl = actionData.roomContextUrls[0];
+
+      if (this._hasPageForUrl(firstContextUrl.location)) {
+        return;
+      }
```

Comment:
I'm not sure this check will be useful since the user is creating the room and he can't add content before the creation, right?